src/com/dotmarketing/viewtools/FileTool.java

```diff
@@ -28,16 +36,20 @@ public void init(Object initData) {
 	public File getNewFile(){
 		return new File();
 	}
-	
+
 	public IFileAsset getFile(String identifier, boolean live){
+		return getFile(identifier, live, -1);
+	}
+
+	public IFileAsset getFile(String identifier, boolean live, long languageId){
 		Identifier id;
 		String p = null;
 		try {
-			id = APILocator.getIdentifierAPI().find(identifier);
+			id = identifierAPI.find(identifier);
 		    if(live){
-			  p = LiveCache.getPathFromCache(id.getURI(), id.getHostId());
+			  p = LiveCache.getPathFromCache(id.getURI(), id.getHostId(), languageId != -1 ? languageId : null);
 		    }else{
-			  p = WorkingCache.getPathFromCache(id.getURI(), id.getHostId());
+			  p = WorkingCache.getPathFromCache(id.getURI(), id.getHostId(), languageId != -1 ? languageId : null);
```

Comment:
The live/working cache calls need to be replaced with `VersionableAPI.getContentletVersionInfo` call.  

Also, if language is <0 we need to call `LanguageWebApi.getLanguage(request)`